content/docs/next-release/performance-tuning.md

```diff
@@ -0,0 +1,91 @@
+
+## Collector settings
+
+### Queue size
+
+Similar to the Agent, the Collector is able to receive spans and place them in an internal queue for processing. This allows the collector to return immediately to the client/agent instead of waiting for the span to make its way to the storage.
+
+The setting `collector.queue-size` (default: `2000`) dictates how many spans the queue should support. In the typical scenario, the queue will be close to empty, as enough workers should exist picking up spans from the queue and sending them to the storage. When the number of items in the queue (metric `jaeger_collector_queue_length`) is permanently high, it’s an indication that either the number of workers should be increased or that the storage cannot keep up with the volume of data that it’s receiving. When the queue is full, the older items in the queue are overridden, causing spans to be discarded (metric `jaeger_collector_spans_dropped_total`).
+
+{{< warning >}}
+The queue size for the agent is about _span batches_, whereas the queue size for the collector is about _individual spans_.
+{{< /warning >}}
+
+Given that the queue size should be close to empty most of the time, this setting should be as high as the available memory for the collector, to provide maximum protection against sudden traffic spikes. However, if your storage layer is under-provisioned and cannot keep up, even a large queue will quickly fill up and start dropping data.
+
+### Workers
+
+Items from the span queue in the collector are picked up by workers. Each worker picks one span from the queue and persists it to the storage. The number of workers can be specified by the setting `collector.num-workers`(default: `50`) and should be as high as needed to keep the queue close to zero. The general rule is: the faster the backing storage, the lower the number of workers can be. Given that workers are relatively cheap, this number can be increased at will. As a general rule, one worker per 50 items in the queue should be sufficient when the storage is fast. With a `collector.queue-size` of `2000`, having about `40` workers should be sufficient. For slower storage mechanisms, this ratio should be adjusted accordingly, having more workers per queue item.
+
+## Deployment architecture
```

Comment:
Shouldn't deployment architecture come first?  I especially think that the section about having the agents close to the applications should come before the Reporter_Flush_Interval section where you talk about how the networking overhead affecting that setting.

Reply:
good point, +1